### PR TITLE
[RKOTLIN-1031] Update to Core 14.4.1

### PR DESCRIPTION
**Pull request by rorbech**
TODO:
- [x] Revisit Core CHANGELOG and duplicate relevant entries. 

Closes: #1682 
Closes: #1707

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,52 @@
+## 1.15.0-SNAPSHOT (YYYY-MM-DD)
+
+[!NOTE]
+This release will bump the Realm file format from version 23 to 24. Opening a file with an older format will automatically upgrade it from fileformat v10. If you want to upgrade from an earlier file format version you will have to use Realm Kotlin v1.13.1 or earlier.  Downgrading to a previous file format is not possible.
```
Comment (clementetb) [inline CHANGELOG.md:4]:
```suggestion
This release will bump the Realm file format from version 23 to 24. Opening a file with an older format will automatically upgrade it from file format v10. If you want to upgrade from an earlier file format version you will have to use Realm Kotlin v1.13.1 or earlier. Downgrading to a previous file format is not possible.
```

**Review by clementetb — APPROVED** [review]
Should we update the changelog issue links to point to Jira? Otherwise, it looks good to me.

File: packages/gradle-plugin/build.gradle.kts [hunk 2]
```diff
@@ -22,6 +23,12 @@ plugins {
     id("realm-publisher")
 }
 
+buildscript {
+    dependencies {
+        classpath("org.yaml:snakeyaml:1.33")
```
Comment (nhachicha) [inline packages/gradle-plugin/build.gradle.kts:28]: Can we store version in `Config#Versions`?

**Review by nhachicha — APPROVED** [review]
Minor things 👍

**Pull request merged**